api/odl.py

```diff
@@ -1133,27 +1092,35 @@ def _detail_for_elementtree_entry(
 
             terms = parser._xpath(odl_license_tag, "odl:terms")
             if terms:
-                total_checkouts = subtag(terms[0], "odl:total_checkouts")
                 concurrent_checkouts = subtag(terms[0], "odl:concurrent_checkouts")
                 expires = subtag(terms[0], "odl:expires")
 
-            license = cls.parse_license(
-                identifier,
-                total_checkouts,
-                concurrent_checkouts,
-                expires,
-                checkout_link,
-                odl_status_link,
-                do_get,
+            if concurrent_checkouts is not None:
+                concurrent_checkouts = int(concurrent_checkouts)
+
+            if expires is not None:
+                expires = to_utc(dateutil.parser.parse(expires))
+
+            if not odl_status_link:
+                continue
+
+            license_info_document = cls.fetch_license_info(odl_status_link, do_get)
+
+            if not license_info_document:
+                continue
+
+            parsed_license = cls.parse_license_info(
+                license_info_document, odl_status_link, checkout_link
             )
 
-            if not license:
+            if not parsed_license:
                 continue
 
-            licenses_owned += int(license.remaining_checkouts or 0)
-            licenses_available += int(license.concurrent_checkouts or 0)
+            assert parsed_license.identifier == identifier
+            assert parsed_license.expires == expires
+            assert parsed_license.terms_concurrency == concurrent_checkouts
```

Comment:
Yes. I wanted to have a visible 💥 if this case ever comes up, because it *should* always be consistent. If its not consistent, its hard to know which value to believe. 

Thinking about it more though, its probably better to log an error, and just skip the license if we see this inconsistency. Does that sound reasonable to you?

Reply:
Yes, I think that would be better. Otherwise, we may have total failures that we don't notice until someone mentions it. Alternatively, we could keep them in and step up monitoring. But I don't think the latter buys us anything practically. We'll get the same error that we could've gotten in the logs. You could add a traceback to the logging if you want to get everything that we'd've seen from the `assert`.

Reply:
Also 💥 !! 😄

Reply:
Updated to log errors.

Reply:
Nice! The logging detail is much more helpful than the tracebacks would've been.